### PR TITLE
IfcElectricFlowTreatmentDevice Rail Import BIMQ 2021-07-21T06:51:56.6

### PR DESCRIPTION
**Pull request by jmirtsch**
Rail properties for IfcElectricFlowTreatmentDevice

### CODE REVIEW

File: IFC4x3/Properties/s/SencondaryFrequency_272XQ9Dt16FOR1$miSL0g$/DocProperty.xml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="SencondaryFrequency_272XQ9Dt16FOR1miSL0g" Name="SencondaryFrequency" UniqueId="870a1689-3770-463d-86c1-ff0b1c540abf" PrimaryDataType="IfcFrequencyMeasure" />
```
Comment (jmirtsch) [inline IFC4x3/Properties/s/SencondaryFrequency_272XQ9Dt16FOR1$miSL0g$/DocProperty.xml:2]:
```suggestion
<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="SecondaryFrequency_272XQ9Dt16FOR1miSL0g" Name="SecondaryFrequency" UniqueId="870a1689-3770-463d-86c1-ff0b1c540abf" PrimaryDataType="IfcFrequencyMeasure" />
```

File: IFC4x3/Properties/s/SencondaryFrequency_272XQ9Dt16FOR1$miSL0g$/Documentation.md [hunk 2]
```diff
@@ -0,0 +1 @@
+The nominal frequency at scondary side
```
Comment (jmirtsch) [inline IFC4x3/Properties/s/SencondaryFrequency_272XQ9Dt16FOR1$miSL0g$/Documentation.md:1]:
```suggestion
The nominal frequency at secondary side
```

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_ElectricFlowTreatmentDeviceTypeElectronicFilter/DocPropertySet.xml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_ElectricFlowTreatmentDeviceTypeElectronicFilter" UniqueId="c6fe5088-1847-ec20-e204-c37a0d269a0d" ApplicableType="IfcElectricFlowTreatmentDevice/ELECTRONICFILTER" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
+	<Properties>
+		<DocProperty xsi:nil="true" href="NominalCurrent_1Sia_y4DvB595bUBLM1Dsb" />
+		<DocProperty xsi:nil="true" href="NominalPower_2pu16hiH1CqQv04BNgxAph" />
+		<DocProperty xsi:nil="true" href="RatedVoltage_0KkDpffYz8L86gzG31TCSi" />
+		<DocProperty xsi:nil="true" href="PrimaryFrequency_3bDvHjEfEAwV9_ElCeTdI" />
+		<DocProperty xsi:nil="true" href="SencondaryFrequency_272XQ9Dt16FOR1miSL0g" />
```
Comment (jmirtsch) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_ElectricFlowTreatmentDeviceTypeElectronicFilter/DocPropertySet.xml:8]:
```suggestion
		<DocProperty xsi:nil="true" href="SecondaryFrequency_272XQ9Dt16FOR1miSL0g" />
```

File: IFC4x3/Properties/t/Type_1CfK$TChb3WvEVYC2jfIpk/DocProperty.xml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="Type_1CfKTChb3WvEVYC2jfIpk" Name="Type" UniqueId="4ca54fdd-32b9-4383-939f-88c0ada52cee" PropertyType="p_enumeratedvalue" PrimaryDataType="IfcLabel">
```
Comment (jmirtsch) [inline IFC4x3/Properties/t/Type_1CfK$TChb3WvEVYC2jfIpk/DocProperty.xml:2]:
```suggestion
<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="FiltersType_1CfKTChb3WvEVYC2jfIpk" Name="FiltersType" UniqueId="4ca54fdd-32b9-4383-939f-88c0ada52cee" PropertyType="p_enumeratedvalue" PrimaryDataType="IfcLabel">
```

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_ElectricFlowTreatmentDeviceTypeElectronicFilter/DocPropertySet.xml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+		<DocProperty xsi:nil="true" href="Type_1CfKTChb3WvEVYC2jfIpk" />
```
Comment (jmirtsch) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_ElectricFlowTreatmentDeviceTypeElectronicFilter/DocPropertySet.xml:9]:
```suggestion
		<DocProperty xsi:nil="true" href="FiltersType_1CfKTChb3WvEVYC2jfIpk" />
```

File: IFC4x3/Properties/n/NominalCurrent_1Sia_y4DvB595bUBLM1Dsb/DocProperty.xml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="NominalCurrent_1Sia_y4DvB595bUBLM1Dsb" Name="NominalCurrent" UniqueId="5cb24fbc-10de-4b14-9165-78b55604dda5" PrimaryDataType="IfcElectricCurrentMeasure" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/n/NominalCurrent_1Sia_y4DvB595bUBLM1Dsb/DocProperty.xml:2]: Reuse

File: IFC4x3/Properties/n/NominalPower_2pu16hiH1CqQv04BNgxAph/DocProperty.xml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="NominalPower_2pu16hiH1CqQv04BNgxAph" Name="NominalPower" UniqueId="b3e011ab-b110-4cd1-ae40-10b5eaecaceb" PrimaryDataType="IfcPowerMeasure" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/n/NominalPower_2pu16hiH1CqQv04BNgxAph/DocProperty.xml:2]: Reuse

File: IFC4x3/Properties/p/PrimaryFrequency_3bDvH$jEfEAwV9_ElCeTdI/DocProperty.xml [hunk 8]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="PrimaryFrequency_3bDvHjEfEAwV9_ElCeTdI" Name="PrimaryFrequency" UniqueId="e537947f-b4ea-4e2b-a7c9-f8ebcca1d9d2" PrimaryDataType="IfcFrequencyMeasure" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/p/PrimaryFrequency_3bDvH$jEfEAwV9_ElCeTdI/DocProperty.xml:2]: Reuse

File: IFC4x3/Properties/r/RatedVoltage_0KkDpffYz8L86gzG31TCSi/DocProperty.xml [hunk 9]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="RatedVoltage_0KkDpffYz8L86gzG31TCSi" Name="RatedVoltage" UniqueId="14b8dce9-a62f-4854-81aa-f500c174c72c" PrimaryDataType="IfcElectricVoltageMeasure" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/r/RatedVoltage_0KkDpffYz8L86gzG31TCSi/DocProperty.xml:2]: Reuse

File: IFC4x3/Properties/s/SencondaryFrequency_272XQ9Dt16FOR1$miSL0g$/DocProperty.xml [hunk 10]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="SecondaryFrequency_272XQ9Dt16FOR1miSL0g" Name="SecondaryFrequency" UniqueId="870a1689-3770-463d-86c1-ff0b1c540abf" PrimaryDataType="IfcFrequencyMeasure" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/s/SencondaryFrequency_272XQ9Dt16FOR1$miSL0g$/DocProperty.xml:2]: Reuse

File: IFC4x3/Properties/t/Type_1CfK$TChb3WvEVYC2jfIpk/DocProperty.xml [hunk 11]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="FiltersType_1CfKTChb3WvEVYC2jfIpk" Name="FiltersType" UniqueId="4ca54fdd-32b9-4383-939f-88c0ada52cee" PropertyType="p_enumeratedvalue" PrimaryDataType="IfcLabel">
```
Comment (SergejMuhic) [inline IFC4x3/Properties/t/Type_1CfK$TChb3WvEVYC2jfIpk/DocProperty.xml:2]: Are there many filters? I would prefer `FilterType` even if there are many.

**Review by SergejMuhic — CHANGES_REQUESTED** [review]
- constant documentation
- pednum documentation

File: IFC4x3/Properties/n/NominalPower_2pu16hiH1CqQv04BNgxAph/DocProperty.xml (hunk 7 above)
Comment (fhulin) [reply IFC4x3/Properties/n/NominalPower_2pu16hiH1CqQv04BNgxAph/DocProperty.xml]: There is no existing generic NominalPower property. So keep the one we add with a generic definition from IEC.

File: IFC4x3/Properties/n/NominalCurrent_1Sia_y4DvB595bUBLM1Dsb/DocProperty.xml (hunk 6 above)
Comment (fhulin) [reply IFC4x3/Properties/n/NominalCurrent_1Sia_y4DvB595bUBLM1Dsb/DocProperty.xml]: Done. Commit: cc2999beef494e17b3c8aa19f47045f89c8c1bb8

File: IFC4x3/Properties/p/PrimaryFrequency_3bDvH$jEfEAwV9_ElCeTdI/DocProperty.xml (hunk 8 above)
Comment (fhulin) [reply IFC4x3/Properties/p/PrimaryFrequency_3bDvH$jEfEAwV9_ElCeTdI/DocProperty.xml]: Done : cc2999beef494e17b3c8aa19f47045f89c8c1bb8

File: IFC4x3/Properties/r/RatedVoltage_0KkDpffYz8L86gzG31TCSi/DocProperty.xml (hunk 9 above)
Comment (fhulin) [reply IFC4x3/Properties/r/RatedVoltage_0KkDpffYz8L86gzG31TCSi/DocProperty.xml]: Done. Commit : cc2999beef494e17b3c8aa19f47045f89c8c1bb8

File: IFC4x3/Properties/s/SencondaryFrequency_272XQ9Dt16FOR1$miSL0g$/DocProperty.xml (hunk 10 above)
Comment (fhulin) [reply IFC4x3/Properties/s/SencondaryFrequency_272XQ9Dt16FOR1$miSL0g$/DocProperty.xml]: Done. Commit : cc2999beef494e17b3c8aa19f47045f89c8c1bb8

File: IFC4x3/Properties/t/Type_1CfK$TChb3WvEVYC2jfIpk/DocProperty.xml (hunk 11 above)
Comment (fhulin) [reply IFC4x3/Properties/t/Type_1CfK$TChb3WvEVYC2jfIpk/DocProperty.xml]: Agree! Commit : cc2999beef494e17b3c8aa19f47045f89c8c1bb8

File: IFC4x3/Properties/n/NominalPower_2pu16hiH1CqQv04BNgxAph/DocProperty.xml (hunk 7 above)
Comment (fhulin) [reply IFC4x3/Properties/n/NominalPower_2pu16hiH1CqQv04BNgxAph/DocProperty.xml]: > * constant documentation
> 
>     * pednum documentation

Documentation is done. Commit : cc2999beef494e17b3c8aa19f47045f89c8c1bb8

Comment (fhulin) [reply IFC4x3/Properties/n/NominalPower_2pu16hiH1CqQv04BNgxAph/DocProperty.xml]: Already using the same NominalPower that have been merged

File: IFC4x3/Properties/n/NominalCurrent_1Sia_y4DvB595bUBLM1Dsb/DocProperty.xml (hunk 6 above)
Comment (fhulin) [reply IFC4x3/Properties/n/NominalCurrent_1Sia_y4DvB595bUBLM1Dsb/DocProperty.xml]: Erased in commit : 7215357c22b15b8f871b30c0abe5495916bdbcf9

**Comment by fhulin** [comment]
@SergejMuhic All issues are now solved.

**Review by AlexBrad1eyCT — APPROVED** [review]
LFTM!

**Review by SergejMuhic — APPROVED** [review]

**Pull request merged**